packages/main/src/plugin/kubernetes/contexts-manager.ts

```diff
@@ -870,7 +928,7 @@ export class ContextsManager {
     return this.states.getCurrentContextGeneralState(this.kubeConfig.currentContext);
   }
 
-  public registerGetCurrentContextResources(resourceName: ResourceName): KubernetesObject[] {
+  public registerGetCurrentContextResources(resourceName: ResourceName): KubernetesObject[] | CoreV1Event[] {
```

Comment:
same remark, maybe we should keep a difference between the KubernetesObject and the events

Reply:
I do not know what is the right approach, as I am not familiar enough with the code, but do we really need to collect all events ? and store them all ?

Shouldn't we do something like logs, and only fetch them for a specific resources, when we need them ? I don't know, but since we do not display all kubernetes resources, if we have events we don't use, that's a bit sad to store them all

Reply:
The advantage of the current method is that the events list is refreshed as soon as new events happen, it can be very helpful when debugging.

There are some filtering to store only Deployments related events:
```
 onAdd: obj => {
        this.states.setStateAndDispatch(context.name, {
          currentContext: this.kubeConfig.currentContext,
          resources: { events: true },
          update: state => {
            if (obj.involvedObject.kind === 'Deployment') {
              state.resources.events.push(obj);
            }
          },
        });
      },
```

Reply:
We will want to extend this to other resources I guess ? I agree with the advantages, but still think it should be a ipc on its own, as this would make more complicated to have two object as result each time :/

Reply:
I have updated the commit with changes, with the events handled separately from the other resources, and with a specific IPC for events. 

I'll have to double-check and test everything, but this is a first step to have the big picture.

Do you think it is ok this way?

Reply:
Finally, I changed back to consider events as a normal secondary resource, as the CoreV1Event is a superset of KubernetesObject